### PR TITLE
Split PixelData into memory and filebacked

### PR DESCRIPTION
**Pull request by oerc0122**
Perform a split of the single `PixelData` object which handles both memory-backed and file-backed in one place into `PixelDataMemory` and `PixelDataFileBacked`, both derived from `PixelDataBase`. 

This allows the user to manipulate pix more precisely and also allows developers to independently optimise and change functions without worrying about what might happen to the memory-backed processes. 

Also makes `PixelDataBase.create` choose memory vs. filebacked based on the `hor_config` setting `mem_chunk_size` to unify page sizes across the Horace code.

### CODE REVIEW

**Review by cmarooney-stfc — CHANGES_REQUESTED** [review]

File: horace_core/sqw/PixelData/@PixelDataMemory/equal_to_tol.m [hunk 1]
```diff
@@ -0,0 +1,114 @@
+function [ok, mess] = equal_to_tol(obj, other_pix, varargin)
+%% EQUAL_TO_TOL Check if two PixelData objects are equal to a given tolerance
```
[the remark this answers is not in the document]
Comment (oerc0122) [reply horace_core/sqw/PixelData/@PixelDataMemory/equal_to_tol.m:2]: This also checks `PixelDataMemory` vs `PixelDataFileBacked`

File: horace_core/sqw/PixelData/@PixelDataMemory/get_pixels.m [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+%
+% Input:
+% ------
+%   abs_pix_indices  A vector of positive integers or a vector of logicals.
+%                    The syntax for these indices attempts to replicate indexing
+%                    into a regular Matlab array. You can use logical indices
+%                    as well as normal indices, and you can index into the array
+%                    "out-of-order". However, you cannot use `end`, but it is
+%                    possible to achieve the same effect using the `num_pixels`
+%                    property.
```
[the remark this answers is not in the document]
Comment (oerc0122) [reply horace_core/sqw/PixelData/@PixelDataMemory/get_pixels.m:24]: What do you mean by "reduce the volume to 2"? This is standard matlab indexing as far as I can tell, with the caveat that you can't use `end`.

File: horace_core/sqw/file_io/@faccess_sqw_v3/private/get_sqw_footer_.m [hunk 3]
```diff
@@ -54,20 +54,29 @@
...
 
-fseek(obj.file_id_,-foot_sz-4,'eof');   % move to start of the block of data (8-byte position + n-byte string + 4-byte string length)
-test_error(obj.file_id_,'Unable to move to the location of the sqw_v3 file descriptor. ErrorMessave: %s')
+try
+    do_fseek(obj.file_id_,-foot_sz-4,'eof');   % move to start of the block of data (8-byte position + n-byte string + 4-byte string length)
+catch ME
+    exc = MException('HORACE:faccess_sqw_v3:io_error',...
+                     'Unable to move to the position of the sqw_v3 file descriptor size.');
+    throw(exc.addCause(ME))
+end
```
Comment (abuts) [inline horace_core/sqw/file_io/@faccess_sqw_v3/private/get_sqw_footer_.m:75]: This is crap -- the previous way was much better

**Review by abuts — APPROVED** [review]
Understanding this in details needs much more involvement than I am ready to do.
I do not like a lot of this, including unwrapping error message throwing, which now takes 7 rows of identical code in different places and keeping the class as a handle. 

But as this is clearly step in the right direction, it should be done ASAP

**Pull request closed without merging**